File: Python/Sorting/insertionsort.py
def insertionSort(arr):
    length = len(arr)

    for x in range(length):
        for y in range(x):
            if arr[y] > arr[x]:
                val = arr.pop(x);
                arr.insert(y, val);

    return arr;


arr = [5, 10, 1, 9, 2, 8, 3, 7, 4, 6, 1];
print(insertionSort(arr));

File: Javascript/insertionsort.js
const insertionSort = arr => {

  let length = arr.length;

  for (var i = 1; i < length; i++) {
    for (var j = 0; j < i; j++) {
      if(arr[i] < arr[j]) {
        // [arr[i], arr[j]] = [arr[j], arr[i]];
        let [val] = arr.splice(i, 1);
        arr.splice(j, 0, val);
      }
    }
  }

  return arr;
}


const numbers = [8, 5, 6, 9, 3, 1, 4, 2, 7, 10];
console.log(insertionSort(numbers));
